homeassistant/components/climate/__init__.py

```diff
@@ -413,12 +413,16 @@ def state_attributes(self):
         """Return the optional state attributes."""
         data = {
             ATTR_CURRENT_TEMPERATURE:
-            self._convert_for_display(self.current_temperature),
+                self._convert_for_display(self.current_temperature),
             ATTR_MIN_TEMP: self._convert_for_display(self.min_temp),
             ATTR_MAX_TEMP: self._convert_for_display(self.max_temp),
-            ATTR_TEMPERATURE:
-            self._convert_for_display(self.target_temperature),
         }
+
+        target_temperature = self.target_temperature
+        if target_temperature is not None:
```

Comment:
Hmm, I didn't know the rule that only temperature controllable things could be climate platforms. This contradicts the climate documentation:

The climate component is built for the controlling and monitoring of HVAC (heating, ventilating, and air conditioning) and thermostat devices.

I will try to find some time to move over to the fan component, although it's unfortunate that that component doesn't support the temperature view.